2-multitenant/README.md

```diff
@@ -1,26 +1,200 @@
+    ]
+  },
+  production = {
+    billing_account    = "111111-111111-111111"
+    folder_id          = "4567890123456"
+    network_project_id = "shared-vpc-project-prod"
+    network_self_link  = "https://www.googleapis.com/compute/v1/projects/shared-vpc-project-prod/global/networks/the-network"
+    org_id             = "2345678901234"
+    subnets_self_links = [
+      "https://www.googleapis.com/compute/v1/projects/shared-vpc-project-prod/regions/us-central1/subnetworks/subnet-us-central1",
+      "https://www.googleapis.com/compute/v1/projects/shared-vpc-project-prod/regions/us-east1/subnetworks/subnet-us-east1"
+    ]
+  }
+}
+```
+
+### Application
+
+For each application to be deployed in this solution, please provide:
+
+- A distinct application name as key;
+  - A distinct acronym for each application, it will be used as a shorten identificator;
+  - IP addresses for each application, when needed;
+  - Certificates for each application, when needed;
+
+```hcl
+apps = {
+  "myapp1" = {
+    acronym          = "MA1"
+    ip_address_names = ["ip-myapp1-01", "ip-myapp1-02"]
+    certificates = {
+      "cert-web" = [
+        "-----BEGIN CERTIFICATE-----",
+        "...",
+        "-----END CERTIFICATE-----"
+      ],
+      "cert-db" = [
+        "-----BEGIN CERTIFICATE-----",
+        "...",
+        "-----END CERTIFICATE-----"
+      ]
+    }
+  }
+
+  "anotherapp" = {
+    acronym = "AAP"
+    # ip_address_names and certificates are optional
+  }
+
+  "yet-another-app" = {
+    acronym          = "YAA"
+    ip_address_names = [] # Empty list example
+    certificates     = {} # Empty map example
+  }
+}
+```
+
+### VPC-SC
+
+This solution can be deployed inside of a VPC-Perimeter. However, the Cloud Build project, aka. seed project, __cannot__ be inside of the perimeter, since it will create new projects, errors will happen when accessing services (enabling APIs for example) before the new project is appended to the perimeter.
+
+__The creation of the Service Perimeter and Access Level are not responsability of this module. But it will make changes, adding projects at the Service Perimeter, creating Directional Rules and adding identities to the Access Level.__
+
+You need to provide an already created Access Level Name, Service Perimeter Name and the mode os deploymente (DRY_RUN or ENFORCED).
+
+This module will modify you perimeter and access level:
+
+- Append Cluster projects at the service perimeter
+- Add Egress rule to allow egress from Network project to Cluster project
+- Add cluster service account, default compute engine and container engine service agent at the access level.
+
+#### Private Workerpool project
+
+If you are deploying this solution inside of the VPC-SC, you need to provide the project where you Private Workerpool is deployed. This module will create Directional rules to allow Egress from Cloud Deploy to the Private Workerpool project.
```

Comment:
does the worker pool project needs to be different from the cloud build project? can it be in the perimeter ?